### PR TITLE
feat(features): add support for Cloud Formation

### PR DESCRIPTION
**Pull request by ferwguerra**

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -6145,6 +6145,7 @@ hal config features edit [parameters]
  * `--artifacts`: Enable artifact support. Read more at [https://spinnaker.io/reference/artifacts/](https://spinnaker.io/reference/artifacts/)
  * `--artifacts-rewrite`: Enable new artifact support. Read more at [https://www.spinnaker.io/reference/artifacts-with-artifactsrewrite/](https://www.spinnaker.io/reference/artifacts-with-artifactsrewrite/)
  * `--chaos`: Enable Chaos Monkey support. For this to work, you'll need a running Chaos Monkey deployment. Currently, Halyard doesn't configure Chaos Monkey for you; read more instructions here [https://github.com/Netflix/chaosmonkey/wiki](https://github.com/Netflix/chaosmonkey/wiki).
+ * `--cloud-formation`: Enable Cloud Formation support.
```
Comment (gregjones) [inline docs/commands.md]: supernit, but "CloudFormation" for consistency in the description

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditFeaturesCommand.java [hunk 2]
```diff
@@ -102,6 +102,12 @@
       arity = 1)
   private Boolean gremlin = null;
 
+  @Parameter(
+      names = "--cloud-formation",
+      description = "Enable Cloud Formation support.",
```
Comment (gregjones) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditFeaturesCommand.java]: "CloudFormation" here too

**Comment by xavileon** [comment]
@spinnaker/oss-approvers PTAL

**Comment by xavileon** [comment]
Hello @spinnaker/oss-approvers, can we have some feedback wether this is a valid approach for enabling the CFN feature through halyard?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverProfileFactory.java [hunk 3]
```diff
@@ -85,6 +80,14 @@ protected void setProfile(
     if (deploymentConfiguration.getProviders() != null) {
       processProviders(deploymentConfiguration.getProviders());
     }
+    Features features = deploymentConfiguration.getFeatures();
```
Comment (ezimanyi) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverProfileFactory.java]: It looks like this PR is creating two places in the hal config to set whether cloudformation is enabled:
* `features.cloudFormation`
* `providers.aws.features.cloudFormation`

The second of these (`providers.aws.features.cloudFormation`) is where we actually want the setting to live as it's where it ends up in the clouddriver config output; this code here looks to be looking at the first location and updating the second to match it.

I don't think we need the `features.cloudFormation` at all; I think that the single source of truth for the value of the setting should be `providers.aws.features.cloudFormation`, which removes the need for this code here and avoids confusingly having two places in the hal config to set it.

In addition to removing the field `features.cloudFormation`, I think you'll also need to change where you've added the command; this should probably be something like `hal config provider aws features ...` and should be registered in `AwsCommand.java`.

Comment (ferwguerra) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverProfileFactory.java]: @ezimanyi Thanks for the feedback!
I agree with you on the changes to be made.
As soon as I have time I will implement them. Thanks again!

Comment (ferwguerra) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverProfileFactory.java]: @ezimanyi could you take a look to the new solution?  Thank you!

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/AbstractEditFeaturesProviderCommand.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google, Inc.
```
Comment (ezimanyi) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/AbstractEditFeaturesProviderCommand.java]: This should reference you or your company (and year should be updated).

Comment (ferwguerra) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/AbstractEditFeaturesProviderCommand.java]: @ezimanyi Thank you for the feedback. Changes done :)

**Pull request merged**